src/test/java/io/xpring/xrpl/DefaultXpringClientTest.java

```diff
@@ -116,20 +122,97 @@ public void getBalanceTestWithFailedAccountInfo() throws IOException, XpringKitE
         // GIVEN a XpringClient with mocked networking which will fail to retrieve account info.
         GRPCResult<GetAccountInfoResponse> accountInfoResult = GRPCResult.error(GENERIC_ERROR);
         DefaultXpringClient client = getClient(
-                accountInfoResult
+                accountInfoResult,
+                GRPCResult.ok(makeTransactionStatus(true, TRANSACTION_STATUS_SUCCESS))
         );
 
         // WHEN the balance is retrieved THEN an error is thrown.
         expectedException.expect(Exception.class);
         client.getBalance(XRPL_ADDRESS);
     }
 
+    @Test
+    public void transactionStatusWithUnvalidatedTransactionAndFailureCode() throws IOException, XpringKitException {
+        // GIVEN a XpringClient which will return an invalidated transaction with a failed code.
+        DefaultXpringClient client = getClient(
+                GRPCResult.ok(makeGetAccountInfoResponse(DROPS_OF_XRP_IN_ACCOUNT)),
+                GRPCResult.ok(makeTransactionStatus(false, TRANSACTION_STATUS_FAILURE))
+        );
+
+        // WHEN the transaction status is retrieved.
+        io.xpring.xrpl.TransactionStatus transactionStatus = client.getTransactionStatus(TRANSACTION_HASH);
+
+        // THEN the status is PENDING.
+        assertThat(transactionStatus).isEqualTo(io.xpring.xrpl.TransactionStatus.PENDING);
+    }
+
+    @Test
+    public void transactionStatusWithUnvalidatedTransactionAndSuccessCode() throws IOException, XpringKitException {
+        // GIVEN a XpringClient which will return an unvalidated transaction with a success code.
+        DefaultXpringClient client = getClient(
+                GRPCResult.ok(makeGetAccountInfoResponse(DROPS_OF_XRP_IN_ACCOUNT)),
+                GRPCResult.ok(makeTransactionStatus(false, TRANSACTION_STATUS_SUCCESS))
+        );
+
+        // WHEN the transaction status is retrieved.
+        io.xpring.xrpl.TransactionStatus transactionStatus = client.getTransactionStatus(TRANSACTION_HASH);
+
+        // THEN the status is PENDING.
+        assertThat(transactionStatus).isEqualTo(io.xpring.xrpl.TransactionStatus.PENDING);
+    }
+
+    @Test
+    public void transactionStatusWithValidatedTransactionAndFailureCode() throws IOException, XpringKitException {
```

Comment:
Should we add a test for a transaction that would be `UNKNOWN` but would currently result in returning a `FAILED` code?

That way we're documenting the current behavior, while we have a TODO, and a test to refactor when we come back to implementing that.

Reply:
Agree we should test this. This is kinda a messy and large thing so I split into a separate PR.

I'm more than happy to land this, but the code will have to make assumptions about the old protocol buffers and there's quite a few TODOs. I figured it would be better to haggle on it separately, but I don't mind merging it into this review, if you'd like. 

Here's the code:
https://github.com/xpring-eng/xpring4j/pull/88